purchase_operating_unit/models/procurement.py

```diff
@@ -10,18 +10,20 @@
 class ProcurementOrder(models.Model):
     _inherit = 'procurement.order'
 
-    @api.one
+    @api.multi
     @api.constrains('purchase_line_id')
     def _check_purchase_order_operating_unit(self):
-        purchase = self.purchase_line_id.order_id
-        location = self.location_id
-        if (purchase and
-                purchase.operating_unit_id != location.operating_unit_id):
-            raise ValidationError(
-                _('Configuration error\nThe Quotation / Purchase Order and '
-                  'the Procurement Order must belong to the same '
-                  'Operating Unit.')
-                )
+        for proc in self:
+            purchase_ou = proc.purchase_line_id.order_id.operating_unit_id
+            location_ou = proc.location_id.operating_unit_id
+            usage = proc.location_id.usage
+            if usage not in ('supplier', 'customer'):
+                if purchase_ou != location_ou:
```

Comment:
There is no point to this embedded if, the code is a single branch, so better as single if.  

The above three lines could be simply written as
```python
internal_usage = proc.location_id.usage not in ('supplier', 'customer')
if internal_usage and purchase_ou != location_ou:
```

But these are all minor points.  Code does what it says.  Its life is ending now anyway as no procurement model in v11 so not blocking comments.